{"cells":[{"cell_type":"code","execution_count":20,"metadata":{},"outputs":[{"output_type":"error","ename":"SyntaxError","evalue":"invalid syntax (<ipython-input-20-51d2fdde5e8c>, line 1)","traceback":["\u001b[1;36m  File \u001b[1;32m\"<ipython-input-20-51d2fdde5e8c>\"\u001b[1;36m, line \u001b[1;32m1\u001b[0m\n\u001b[1;33m    This repository contains custom pipes and models related to using spaCy for scientific documents.\u001b[0m\n\u001b[1;37m                  ^\u001b[0m\n\u001b[1;31mSyntaxError\u001b[0m\u001b[1;31m:\u001b[0m invalid syntax\n"]}],"source":["This repository contains custom pipes and models related to using spaCy for scientific documents.\n","\n","In particular, there is a custom tokenizer that adds tokenization rules on top of spaCy's rule-based tokenizer, a POS tagger and syntactic parser trained on biomedical data and an entity span detection model. Separately, there are also NER models for more specific tasks.\n","\n","An NER System is capable of discovering entity elements from raw data and determines the category the element belongs to. The system reads the sentence and highlights the important entity elements in the text. NER might be given separate sensitive entities depending on the project. This means that NER systems designed for one project may not be reused for another task.\n","\n","https://github.com/allenai/scispacy"]},{"metadata":{"trusted":true},"cell_type":"code","source":["!pip install scispacy\n","!pip install https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/releases/v0.2.4/en_core_sci_sm-0.2.4.tar.gz"],"execution_count":8,"outputs":[{"output_type":"stream","name":"stdout","text":"Requirement already satisfied: scispacy in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (0.2.4)\nRequirement already satisfied: scikit-learn>=0.20.3 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from scispacy) (0.22.1)\nRequirement already satisfied: nmslib>=1.7.3.6 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from scispacy) (2.0.5)\nRequirement already satisfied: pysbd in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from scispacy) (0.2.3)\nRequirement already satisfied: joblib in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from scispacy) (0.14.1)\nRequirement already satisfied: numpy in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from scispacy) (1.18.1)\nRequirement already satisfied: conllu in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from scispacy) (2.3.2)\nRequirement already satisfied: spacy>=2.2.1 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from scispacy) (2.2.4)\nRequirement already satisfied: awscli in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from scispacy) (1.18.28)\nRequirement already satisfied: scipy>=0.17.0 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from scikit-learn>=0.20.3->scispacy) (1.4.1)\nRequirement already satisfied: pybind11>=2.2.3 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from nmslib>=1.7.3.6->scispacy) (2.4.3)\nRequirement already satisfied: psutil in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from nmslib>=1.7.3.6->scispacy) (5.6.7)\nRequirement already satisfied: requests<3.0.0,>=2.13.0 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->scispacy) (2.22.0)\nRequirement already satisfied: plac<1.2.0,>=0.9.6 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->scispacy) (1.1.3)\nRequirement already satisfied: tqdm<5.0.0,>=4.38.0 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->scispacy) (4.42.0)\nRequirement already satisfied: blis<0.5.0,>=0.4.0 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->scispacy) (0.4.1)\nRequirement already satisfied: cymem<2.1.0,>=2.0.2 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->scispacy) (2.0.3)\nRequirement already satisfied: srsly<1.1.0,>=1.0.2 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->scispacy) (1.0.2)\nRequirement already satisfied: thinc==7.4.0 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->scispacy) (7.4.0)\nRequirement already satisfied: wasabi<1.1.0,>=0.4.0 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->scispacy) (0.6.0)\nRequirement already satisfied: catalogue<1.1.0,>=0.0.7 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->scispacy) (1.0.0)\nRequirement already satisfied: preshed<3.1.0,>=3.0.2 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->scispacy) (3.0.2)\nRequirement already satisfied: setuptools in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->scispacy) (41.4.0)\nRequirement already satisfied: murmurhash<1.1.0,>=0.28.0 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->scispacy) (1.0.2)\nRequirement already satisfied: s3transfer<0.4.0,>=0.3.0 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from awscli->scispacy) (0.3.3)\nRequirement already satisfied: docutils<0.16,>=0.10 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from awscli->scispacy) (0.15.2)\nRequirement already satisfied: botocore==1.15.28 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from awscli->scispacy) (1.15.28)\nRequirement already satisfied: PyYAML<5.4,>=3.10; python_version != \"3.4\" in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from awscli->scispacy) (5.1.2)\nRequirement already satisfied: colorama<0.4.4,>=0.2.5; python_version != \"3.4\" in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from awscli->scispacy) (0.4.3)\nRequirement already satisfied: rsa<=3.5.0,>=3.1.2 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from awscli->scispacy) (3.4.2)\nRequirement already satisfied: certifi>=2017.4.17 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from requests<3.0.0,>=2.13.0->spacy>=2.2.1->scispacy) (2019.11.28)\nRequirement already satisfied: idna<2.9,>=2.5 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from requests<3.0.0,>=2.13.0->spacy>=2.2.1->scispacy) (2.8)\nRequirement already satisfied: chardet<3.1.0,>=3.0.2 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from requests<3.0.0,>=2.13.0->spacy>=2.2.1->scispacy) (3.0.4)\nRequirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from requests<3.0.0,>=2.13.0->spacy>=2.2.1->scispacy) (1.24.2)\nRequirement already satisfied: importlib-metadata>=0.20; python_version < \"3.8\" in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from catalogue<1.1.0,>=0.0.7->spacy>=2.2.1->scispacy) (1.4.0)\nRequirement already satisfied: jmespath<1.0.0,>=0.7.1 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from botocore==1.15.28->awscli->scispacy) (0.9.5)\nRequirement already satisfied: python-dateutil<3.0.0,>=2.1 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from botocore==1.15.28->awscli->scispacy) (2.8.1)\nRequirement already satisfied: pyasn1>=0.1.3 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from rsa<=3.5.0,>=3.1.2->awscli->scispacy) (0.4.8)\nRequirement already satisfied: zipp>=0.5 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from importlib-metadata>=0.20; python_version < \"3.8\"->catalogue<1.1.0,>=0.0.7->spacy>=2.2.1->scispacy) (0.6.0)\nRequirement already satisfied: six>=1.5 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from python-dateutil<3.0.0,>=2.1->botocore==1.15.28->awscli->scispacy) (1.12.0)\nRequirement already satisfied: more-itertools in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from zipp>=0.5->importlib-metadata>=0.20; python_version < \"3.8\"->catalogue<1.1.0,>=0.0.7->spacy>=2.2.1->scispacy) (8.1.0)\nCollecting https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/releases/v0.2.4/en_core_sci_sm-0.2.4.tar.gz\n  Using cached https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/releases/v0.2.4/en_core_sci_sm-0.2.4.tar.gz\nRequirement already satisfied (use --upgrade to upgrade): en-core-sci-sm==0.2.4 from https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/releases/v0.2.4/en_core_sci_sm-0.2.4.tar.gz in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages\nRequirement already satisfied: spacy>=2.2.1 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from en-core-sci-sm==0.2.4) (2.2.4)\nRequirement already satisfied: requests<3.0.0,>=2.13.0 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (2.22.0)\nRequirement already satisfied: preshed<3.1.0,>=3.0.2 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (3.0.2)\nRequirement already satisfied: tqdm<5.0.0,>=4.38.0 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (4.42.0)\nRequirement already satisfied: plac<1.2.0,>=0.9.6 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (1.1.3)\nRequirement already satisfied: cymem<2.1.0,>=2.0.2 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (2.0.3)\nRequirement already satisfied: srsly<1.1.0,>=1.0.2 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (1.0.2)\nRequirement already satisfied: setuptools in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (41.4.0)\nRequirement already satisfied: wasabi<1.1.0,>=0.4.0 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (0.6.0)\nRequirement already satisfied: numpy>=1.15.0 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (1.18.1)\nRequirement already satisfied: blis<0.5.0,>=0.4.0 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (0.4.1)\nRequirement already satisfied: thinc==7.4.0 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (7.4.0)\nRequirement already satisfied: catalogue<1.1.0,>=0.0.7 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (1.0.0)\nRequirement already satisfied: murmurhash<1.1.0,>=0.28.0 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from spacy>=2.2.1->en-core-sci-sm==0.2.4) (1.0.2)\nRequirement already satisfied: certifi>=2017.4.17 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from requests<3.0.0,>=2.13.0->spacy>=2.2.1->en-core-sci-sm==0.2.4) (2019.11.28)\nRequirement already satisfied: idna<2.9,>=2.5 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from requests<3.0.0,>=2.13.0->spacy>=2.2.1->en-core-sci-sm==0.2.4) (2.8)\nRequirement already satisfied: chardet<3.1.0,>=3.0.2 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from requests<3.0.0,>=2.13.0->spacy>=2.2.1->en-core-sci-sm==0.2.4) (3.0.4)\nRequirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from requests<3.0.0,>=2.13.0->spacy>=2.2.1->en-core-sci-sm==0.2.4) (1.24.2)\nRequirement already satisfied: importlib-metadata>=0.20; python_version < \"3.8\" in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from catalogue<1.1.0,>=0.0.7->spacy>=2.2.1->en-core-sci-sm==0.2.4) (1.4.0)\nRequirement already satisfied: zipp>=0.5 in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from importlib-metadata>=0.20; python_version < \"3.8\"->catalogue<1.1.0,>=0.0.7->spacy>=2.2.1->en-core-sci-sm==0.2.4) (0.6.0)\nRequirement already satisfied: more-itertools in c:\\users\\benettim\\appdata\\local\\continuum\\miniconda3\\lib\\site-packages (from zipp>=0.5->importlib-metadata>=0.20; python_version < \"3.8\"->catalogue<1.1.0,>=0.0.7->spacy>=2.2.1->en-core-sci-sm==0.2.4) (8.1.0)\nBuilding wheels for collected packages: en-core-sci-sm\n  Building wheel for en-core-sci-sm (setup.py): started\n  Building wheel for en-core-sci-sm (setup.py): finished with status 'done'\n  Created wheel for en-core-sci-sm: filename=en_core_sci_sm-0.2.4-cp37-none-any.whl size=17161113 sha256=1e68917550364851e3dafcaf1449fd1ab3ad01d5794ecc701435987eb0b94139\n  Stored in directory: C:\\Users\\BenettiM\\AppData\\Local\\pip\\Cache\\wheels\\34\\60\\b9\\fabd9c3eeba17ed66df745479f2fc502a6702755cb4a9632f2\nSuccessfully built en-core-sci-sm\n"}]},{"metadata":{"trusted":true},"cell_type":"code","source":["import pandas as pd\n","import scispacy # https://allenai.github.io/scispacy/   https://github.com/allenai/scispacy  \n","import spacy\n","import numpy as np\n","from tqdm import tqdm\n","from sklearn.metrics.pairwise import cosine_similarity"],"execution_count":9,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":["nlp = spacy.load(\"en_core_sci_sm\")\n","meta = pd.read_csv(\"C:/Users/BenettiM/Downloads/CORD-19-research-challenge/metadata.csv\")"],"execution_count":10,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":["vector_dict = {}\n","for sha, abstract in tqdm(meta[[\"sha\",\"abstract\"]].values):\n","    if isinstance(abstract, str):\n","        vector_dict[sha] = nlp(abstract).vector"],"execution_count":11,"outputs":[{"output_type":"stream","name":"stderr","text":"100%|██████████| 44220/44220 [20:34<00:00, 35.82it/s]\n"}]},{"metadata":{"trusted":true},"cell_type":"code","source":["keys = list(vector_dict.keys())\n","values = list(vector_dict.values())"],"execution_count":12,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":["cosine_sim_matrix = cosine_similarity(values, values)"],"execution_count":13,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":["n_sim_articles = 5\n","input_sha = \"e3b40cc8e0e137c416b4a2273a4dca94ae8178cc\"\n","\n","\n","sha_index = keys.index(input_sha)\n","sim_indexes = np.argsort(cosine_sim_matrix[sha_index])[::-1][1:n_sim_articles+1]\n","sim_shas = [keys[i] for i in sim_indexes]\n","meta_info = meta[meta.sha.isin(sim_shas)]"],"execution_count":14,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":["print(\"-------QUERY ABSTRACT-----\")\n","print(meta[meta.sha == input_sha][\"abstract\"].values[0])"],"execution_count":15,"outputs":[{"output_type":"stream","name":"stdout","text":"-------QUERY ABSTRACT-----\nIn December 2019, cases of unidentified pneumonia with a history of exposure in the Huanan Seafood Market were reported in Wuhan, Hubei Province. A novel coronavirus, SARS-CoV-2, was identified to be accountable for this disease. Human-to-human transmission is confirmed, and this disease (named COVID-19 by World Health Organization (WHO)) spread rapidly around the country and the world. As of 18 February 2020, the number of confirmed cases had reached 75,199 with 2009 fatalities. The COVID-19 resulted in a much lower case-fatality rate (about 2.67%) among the confirmed cases, compared with Severe Acute Respiratory Syndrome (SARS) and Middle East Respiratory Syndrome (MERS). Among the symptom composition of the 45 fatality cases collected from the released official reports, the top four are fever, cough, short of breath, and chest tightness/pain. The major comorbidities of the fatality cases include hypertension, diabetes, coronary heart disease, cerebral infarction, and chronic bronchitis. The source of the virus and the pathogenesis of this disease are still unconfirmed. No specific therapeutic drug has been found. The Chinese Government has initiated a level-1 public health response to prevent the spread of the disease. Meanwhile, it is also crucial to speed up the development of vaccines and drugs for treatment, which will enable us to defeat COVID-19 as soon as possible.\n"}]},{"metadata":{"trusted":true},"cell_type":"code","source":["print(f\"----TOP {n_sim_articles} SIMILAR ABSTRACTS-----\")\n","for abst in meta_info.abstract.values:\n","    print(abst)\n","    print(\"---------\")"],"execution_count":16,"outputs":[{"output_type":"stream","name":"stdout","text":"----TOP 5 SIMILAR ABSTRACTS-----\nAbstract The 29th International Conference on Antiviral Research (ICAR) was held in La Jolla, CA, USA from April 17 to 21, 2016. This report opens with a tribute to the late Chris McGuigan, a Past-President of ISAR, then continues with summaries of the principal invited lectures. Doug Richman (Elion Award) investigated HIV resistance, Bob Vince (Holý Award) showed how carbocyclic nucleoside analogs led to abacavir and Jerome Deval (Prusoff Award) explained how his group chose to seek a nucleoside analog to treat RSV. ALS-8176 was active in a human RSV-challenge study and is being evaluated in children. The first keynote address, by Richard H. Scheuermann, reported on the remarkable progress made in viral genomics. The second keynote address, by Heinz Feldmann, gave an overview of Ebola virus disease. There were four mini-symposia, Structural Biology, Diagnostic Technologies, DNA viruses and Zika virus. Diagnostic assays are approaching an ideal aim, a compact instrument, simple to use with any type of sample, no sample preparation and a result within an hour. The diversity of HCMV is far greater than for other herpesviruses, typically, an individual having >20,000 single nucleotide polymorphisms (SNPs). During antiviral treatment, there is rapid CMV evolution which is presumed to be due to preferential selection of already present variants rather than by the creation of new variants. A selection of contributor presentations includes oral prodrugs for nucleoside triphosphate analogs, a new method for the synthesis of phosphoramidate prodrugs and the clinical evaluation of brincidofovir for treating transplant recipients with adenovirus infections.\n---------\nThe clinical evaluation of convalescent plasma (CP) for the treatment of Ebola virus disease (EVD) in the current outbreak, predominantly affecting Guinea, Sierra Leone, and Liberia, was prioritized by the World Health Organization in September 2014. In each of these countries, nonrandomized comparative clinical trials were initiated. The Ebola-Tx trial in Conakry, Guinea, enrolled 102 patients by 7 July 2015; no severe adverse reactions were noted. The Ebola-CP trial in Sierra Leone and the EVD001 trial in Liberia have included few patients. Although no efficacy data are available yet, current field experience supports the safety, acceptability, and feasibility of CP as EVD treatment. Longer-term follow-up as well as data from nontrial settings and evidence on the scalability of the intervention are required. CP sourced from within the outbreak is the most readily available source of anti-EVD antibodies. Until the advent of effective antivirals or monoclonal antibodies, CP merits further evaluation.\n---------\nSince December 2019, COVID-19 has raged in Wuhan and subsequently all over China and the world. We propose a Cybernetics-based Dynamic Infection Model (CDIM) to the dynamic infection process with a probability distributed incubation delay and feedback principle. Reproductive trends and the stability of the SARS-COV-2 infection in a city can then be analyzed, and the uncontrollable risks can be forecasted before they really happen. The infection mechanism of a city is depicted using the philosophy of cybernetics and approaches of the control engineering. Distinguished with other epidemiological models, such as SIR, SEIR, etc., that compute the theoretical number of infected people in a closed population, CDIM considers the immigration and emigration population as system inputs, and administrative and medical resources as dynamic control variables. The epidemic regulation can be simulated in the model to support the decision-making for containing the outbreak. City case studies are demonstrated for verification and validation.\n---------\nBACKGROUND: Until 2008, human rabies had never been reported in French Guiana. On 28 May 2008, the French National Reference Center for Rabies (Institut Pasteur, Paris) confirmed the rabies diagnosis, based on hemi-nested polymerase chain reaction on skin biopsy and saliva specimens from a Guianan, who had never travelled overseas and died in Cayenne after presenting clinically typical meningoencephalitis. METHODOLOGY/PRINCIPAL FINDINGS: Molecular typing of the virus identified a Lyssavirus (Rabies virus species), closely related to those circulating in hematophagous bats (mainly Desmodus rotundus) in Latin America. A multidisciplinary Crisis Unit was activated. Its objectives were to implement an epidemiological investigation and a veterinary survey, to provide control measures and establish a communications program. The origin of the contamination was not formally established, but was probably linked to a bat bite based on the virus type isolated. After confirming exposure of 90 persons, they were vaccinated against rabies: 42 from the case's entourage and 48 healthcare workers. To handle that emergence and the local population's increased demand to be vaccinated, a specific communications program was established using several media: television, newspaper, radio. CONCLUSION/SIGNIFICANCE: This episode, occurring in the context of a Department far from continental France, strongly affected the local population, healthcare workers and authorities, and the management team faced intense pressure. This observation confirms that the risk of contracting rabies in French Guiana is real, with consequences for population educational program, control measures, medical diagnosis and post-exposure prophylaxis.\n---------\nThe Brighton Collaboration Viral Vector Vaccines Safety Working Group (V3SWG) was formed to evaluate the safety and characteristics of live, recombinant viral vector vaccines. A recent publication by the V3SWG described live, attenuated, recombinant vesicular stomatitis virus (rVSV) as a chimeric virus vaccine for HIV-1 (Clarke et al., 2016). The rVSV vector system is being explored as a platform for development of multiple vaccines. This paper reviews the molecular and biological features of the rVSV vector system, followed by a template with details on the safety and characteristics of a rVSV vaccine against Zaire ebolavirus (ZEBOV). The rVSV-ZEBOV vaccine is a live, replication competent vector in which the VSV glycoprotein (G) gene is replaced with the glycoprotein (GP) gene of ZEBOV. Multiple copies of GP are expressed and assembled into the viral envelope responsible for inducing protective immunity. The vaccine (designated V920) was originally constructed by the National Microbiology Laboratory, Public Health Agency of Canada, further developed by NewLink Genetics Corp. and Merck & Co., and is now in final stages of registration by Merck. The vaccine is attenuated by deletion of the principal virulence factor of VSV (the G protein), which also removes the primary target for anti-vector immunity. The V920 vaccine caused no toxicities after intramuscular (IM) or intracranial injection of nonhuman primates and no reproductive or developmental toxicity in a rat model. In multiple studies, cynomolgus macaques immunized IM with a wide range of virus doses rapidly developed ZEBOV-specific antibodies measured in IgG ELISA and neutralization assays and were fully protected against lethal challenge with ZEBOV virus. Over 20,000 people have received the vaccine in clinical trials; the vaccine has proven to be safe and well tolerated. During the first few days after vaccination, many vaccinees experience a mild acute-phase reaction with fever, headache, myalgia, and arthralgia of short duration; this period is associated with a low-level viremia, activation of anti-viral genes, and increased levels of chemokines and cytokines. Oligoarthritis and rash appearing in the second week occur at a low incidence, and are typically mild-moderate in severity and self-limited. V920 vaccine was used in a Phase III efficacy trial during the West African Ebola epidemic in 2015, showing 100% protection against Ebola Virus Disease, and it has subsequently been deployed for emergency control of Ebola outbreaks in central Africa. The template provided here provides a comprehensive picture of the first rVSV vector to reach the final stage of development and to provide a solution to control of an alarming human disease.\n---------\n"}]},{"metadata":{"trusted":true},"cell_type":"code","source":["n_return = 5\n","query_statement = \"Age-adjusted mortality data for Acute Respiratory Distress Syndrome (ARDS) with/without other organ failure – particularly for viral etiologies\""],"execution_count":17,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":["query_vector = nlp(query_statement).vector\n","cosine_sim_matrix_query = cosine_similarity(values, query_vector.reshape(1,-1))\n","query_sim_indexes = np.argsort(cosine_sim_matrix_query.reshape(1,-1)[0])[::-1][:n_return]\n","query_shas = [keys[i] for i in query_sim_indexes]\n","meta_info_query = meta[meta.sha.isin(query_shas)]"],"execution_count":18,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":["print(f\"----TOP {n_return} SIMILAR ABSTRACTS TO QUERY-----\")\n","for abst in meta_info_query.abstract.values:\n","    print(abst)\n","    print(\"---------\")"],"execution_count":19,"outputs":[{"output_type":"stream","name":"stdout","text":"----TOP 5 SIMILAR ABSTRACTS TO QUERY-----\nRésumé Introduction Le métapneumovirus humain (hMPV) est un virus récemment identifié chez l’homme, responsable d’infections respiratoires parfois sévères que l’on observe surtout chez l’enfant. Observation Un patient âgé de 59 ans a été hospitalisé pour une atteinte respiratoire fébrile 3 jours après le retour d’un voyage en Chine effectué pendant l’épidémie de syndrome respiratoire aigu sévère (Sras). En dehors de la fièvre (>38 °C), étaient notées une toux sèche, des myalgies, des arthralgies, une opacité paracardiaque droite et une lymphopénie modérée. Les recherches virologiques conventionnelles étaient négatives. La recherche du nouveau coronavirus responsable du Sras était négative, mais la recherche de métapneumovirus humain (hMPV) était positive. Discussion Cette observation indique que le hMPV peut être responsable d’une atteinte respiratoire fébrile pouvant initialement évoquer un Sras chez un patient ayant séjourné en zone d’endémie. Conclusion La recherche de hMPV paraît indiquée dans les infections respiratoires aiguës de l’adulte. Summary Introduction Human metapneumovirus (hMPV) is a newly identified virus responsible for occasionally severe respiratory infections, mainly observed in children. Observation A 59-year old man was hospitalised for respiratory problmems and fever 3 days after his return from China during the epidemic of severe acute respiratory syndrome (SARS). Other than fever (> 38°C), the patient exhibited a dry cough, myalgias, arthralgias and mild lymphopenia and the chest radiography showed an opacity on the right lower lung. Conventional virological investigations remained negative. Detection of the novel SARS-associated coronavirus was negative but hMPV detection was positive. Discussion The present observation indicates that hMPV may cause respiratory symptoms mimicking initial manifestations of SARS in adults visiting epidemic countries. Conclusion hMPV testing has clinical utility in adult patients with acute respiratory tract infection.\n---------\nRésumé Contexte Une nouvelle coronavirose a été décrite chez le furet, notamment en Espagne et aux États-Unis, induite par un Coronavirus du groupe 1, proche du Coronavirus responsable de l’entérite catarrhale enzootique et à l’origine de lésions histologiques similaires à celle de la péritonite infectieuse féline (PIF) dans l’espèce féline. Objectif Démontrer l’origine virale de lésions pyogranulomateuses sévères au sein de tissus lymphoïdes ou viscéraux de trois furets. Méthodes Analyse histologique et immunohistochimique de lésions reçues au laboratoire suite à des autopsies ou des laparotomies exploratrices pratiquées par des vétérinaires. Résultats L’analyse histologique révèle une inflammation pyogranulomateuse multicentrique, touchant préférentiellement les nœuds lymphatiques abdominaux et différents organes abdominaux. L’analyse immunohistochimique permet la mise en évidence d’antigènes viraux de Coronavirus en position intramacrophagique au sein de lésions pyogranulomateuses ganglionnaires. Conclusion Ces trois cas d’inflammation pyogranulomateuse multicentrique sont dus à une coronavirose qui présente de nombreux points de ressemblance avec la PIF. Summary Background A disease resembling feline infectious peritonitis (FIP) has been recognized recently in ferrets, mostly in United States and in Spain. This disease is due to a Group 1 Coronavirus, closely related to the ferret enteric Coronavirus. Objective To demonstrate the viral origin of severe pyogranulomatous lymphoid and visceral lesions in three ferrets. Methods Histopathology and immunohistochemistry of lesions received in our laboratory. Results Severe pyogranulomatous inflammation was observed in the abdominal lymph nodes and in various organs. Positive labelling of Coronavirus antigen was demonstrated in macrophages in the pyogranulomatous lesions by immunohistochemistry. Conclusion The three cases of severe pyogranulomatous disseminated inflammation were due to a Coronavirus, causing a systemic coronavirosis in ferret with a high similarity to FIP.\n---------\nSummary Background A stochastic discrete event simulation model was developed to assess the effectiveness of passenger screening for Pandemic Influenza (PI) at U.S. airport foreign entry. Methods International passengers arriving at 18 U.S. airports from Asia, Europe, South America, and Canada were assigned to one of three states: not infected, infected with PI, infected with other respiratory illness. Passengers passed through layered screening then exited the model. 80% screening effectiveness was assumed for symptomatic passengers; 6% asymptomatic passengers. Results In the first 100 days of a global pandemic, U.S. airport screening would evaluate over 17M passengers with 800K secondary screenings. 11,570 PI infected passengers (majority asymptomatic) would enter the U.S. undetected from all 18 airports. Foreign airport departure screening significantly decreased the false negative (infected/undetected) passengers. U.S. attack rates: no screening (26.9%–30.9%); screening (26.4%–30.6%); however airport screening results in 800K–1.8M less U.S. PI cases; 16K–35K less deaths (2% fatality rate). Antiviral medications for travel contact prophylaxis (10 contacts/PI passenger) were high – 8.8M. False positives from all 18 airports: 100–200/day. Conclusions Foreign shore exit screening greatly reduces numbers of PI infected passengers. U.S. airport screening identifies 50% infected individuals; efficacy is limited by the asymptomatic PI infected. Screening will not significantly delay arrival of PI via international air transport, but will reduce the rate of new US cases and subsequent deaths.\n---------\nSummary Background Middle East respiratory syndrome coronavirus (MERS-CoV) is a zoonotic infection causing severe viral pneumonia, with index cases having resided in or recently travelled to the Arabian peninsula, and is a global concern for public health. Limited human-to-human transmission, leading to some case clusters, has been reported. MERS-CoV has been reported in dromedary camels but phenotypic characterisation of such viruses is limited. We aimed to compare MERS-CoV isolates from dromedaries in Saudi Arabia and Egypt with a prototype human MERS-CoV to assess virus replication competence and cell tropism in ex-vivo cultures of human bronchus and lung. Methods We characterised MERS-CoV viruses from dromedaries in Saudi Arabia and Egypt and compared them with a human MERS-CoV reference strain. We assessed viral replication kinetics and competence in Vero-E6 cells (rhesus monkey), tissue tropism in cultures of ex-vivo human bronchial and lung tissues, and cytokine and chemokine induction, gene expression, and quantification of viral RNA in Calu-3 cells (human respiratory tract). We used mock-infected tissue as negative controls for ex-vivo experiments and influenza A H5N1 as a positive control for cytokine and chemokine induction experiments in Calu-3 cells. Findings We isolated three dromedary strains, two from Saudi Arabia (Dromedary/Al-Hasa-KFU-HKU13/2013 [AH13] and Dromedary/Al-Hasa-KFU-HKU19D/2013 [AH19D]), and one from Egypt (Dromedary/Egypt-NRCE-HKU270/2013 [NRCE-HKU270]). The human and dromedary MERS-CoV strains had similar viral replication competence in Vero-E6 cells and respiratory tropism in ex-vivo cultures of the human respiratory tract, and had similar ability to evade interferon responses in the human-respiratory-tract-derived cell line Calu-3. Interpretation The similarity of virus tropism and replication competence of human and dromedary MERS-CoV from the Arabian peninsula, and genetically diverse dromedary viruses from Egypt, in ex-vivo cultures of the human respiratory tract suggests that dromedary viruses from Saudi Arabia and Egypt are probably infectious to human beings. Exposure to zoonotic MERS-CoV is probably occurring in a wider geographical region beyond the Arabian peninsula. Funding King Faisal University, Egyptian National Research Centre, Hong Kong Food and Health Bureau, National Institute of Allergy and Infectious Diseases, and European Community Seventh Framework Program.\n---------\nDuring the severe acute respiratory syndrome (SARS) outbreak, electronic media made it possible to disseminate prevention messages rapidly. The Centers for Disease Control and Prevention’s Travelers’ Health Web site was frequently visited in the first half of 2003; more than 2.6 million visits were made to travel alerts, advisories, and other SARS-related documents.\n---------\n"}]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["# -- Next step is to create an analysis base on the data provided by the document matrix\n","# 25-03-2020"]}],"metadata":{"kernelspec":{"language":"python","display_name":"Python 3","name":"python3"},"language_info":{"pygments_lexer":"ipython3","nbconvert_exporter":"python","version":"3.8.1-final","file_extension":".py","codemirror_mode":{"name":"ipython","version":3},"name":"python","mimetype":"text/x-python"}},"nbformat":4,"nbformat_minor":4}